api/app/signals/apps/search/documents/base.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from django.db.models import Case, When
+from elasticsearch.helpers import bulk
+from elasticsearch_dsl import Document, Index, Search
+
+log = logging.getLogger(__name__)
+
+
+class CustomSearch(Search):
+
+    def __init__(self, **kwargs):
+        self._queryset = kwargs.pop('queryset', None)
```

Comment:
Not sure but when I removed it here or try to set it without the `None` it failed. Did not investigated it after I tried it, maybe we can leave it for now so that we can merge and do the tests and try to fix/improve this in the future?